src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioWorkspaceStatusServiceFactory.cs

```diff
@@ -76,6 +76,9 @@ public Service(IAsyncServiceProvider2 serviceProvider, IAsynchronousOperationLis
                 Task.Run(() => EnsureInitializationAsync(CancellationToken.None), CancellationToken.None).CompletesAsyncOperation(asyncToken);
             }
 
+            // unfortunately, IVsOperationProgressStatusService requires UI thread to let project system to proceed to next stages.
+            // this method should only be used with either await or JTF.Run, it should be never used with Task.Wait otherwise, it can
+            // deadlock
```

Comment:
Move this to a doc comment so there's some chance that somebody will actually see this at a call site.

Reply:
The comment should probably also be moved to the workspace service itself (or some comment like that) since most consumers are going to see it there -- nobody really sees the direct type like this.

Reply:
📝 All asynchronous APIs defined for Visual Studio fall into this category unless explicitly stated otherwise.

Reply:
it is not all async API from VS. synchronous API from VS can also fall into this category. so any call to VS API from async caller that we do "Wait" on UI thread can cause an issue.

and I am not solving that root issue in this PR. if we want to solve that issue, feel free to open an issue where we can discuss how we can embrass JTF more deepfully in Roslyn.